--- dracut-033/modules.d/95resume/module-setup.sh.orig	2013-09-15 18:27:32.700252837 +0200
+++ dracut-033/modules.d/95resume/module-setup.sh	2013-09-15 18:30:44.410244789 +0200
@@ -17,7 +17,7 @@
 install() {
     local _bin
     # Optional uswsusp support
-    for _bin in /usr/sbin/resume /usr/lib/suspend/resume /usr/lib/uswsusp/resume
+    for _bin in /usr/sbin/resume /usr/@lib@/suspend/resume /usr/@lib@/uswsusp/resume
     do
         [[ -x "${_bin}" ]] && {
             inst "${_bin}" /usr/sbin/resume
