--- /dev/null	2012-03-21 19:30:19.123295831 +0100
+++ dracut-017/modules.d/01bash/module-setup.sh	2012-03-21 23:46:14.717134386 +0100
@@ -0,0 +1,18 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+
+check() {
+    return 0
+}
+
+depends() {
+    return 0
+}
+
+install() {
+    # If another shell is already installed, do not use bash
+    [[ -x $initdir/bin/sh ]] && return
+
+    inst /bin/bash && ln -sf bash "${initdir}/bin/sh"
+}
