diff -dur dracut-034.orig/modules.d/98systemd/dracut-initqueue.sh dracut-034/modules.d/98systemd/dracut-initqueue.sh
--- dracut-034.orig/modules.d/98systemd/dracut-initqueue.sh	2013-10-08 09:55:26.000000000 +0200
+++ dracut-034/modules.d/98systemd/dracut-initqueue.sh	2014-01-11 15:02:54.852353296 +0100
@@ -52,6 +52,10 @@
     # no more udev jobs and queues empty.
     sleep 0.5
 
+    for i in /run/systemd/ask-password/ask.*; do
+        [ -e "$i" ] && continue 2
+    done
+
     if [ $main_loop -gt $((2*$RDRETRY/3)) ]; then
         for job in $hookdir/initqueue/timeout/*.sh; do
             [ -e "$job" ] || break
