--- dracut-047/modules.d/50plymouth/plymouth-populate-initrd.sh.orig	2018-02-19 19:30:07.910250308 +0100
+++ dracut-047/modules.d/50plymouth/plymouth-populate-initrd.sh	2018-02-19 19:31:04.133205683 +0100
@@ -4,7 +4,7 @@
 PLYMOUTH_THEME=$(plymouth-set-default-theme)
 
 inst_multiple plymouthd plymouth \
-    /etc/system-release
+    /etc/os-release
 
 test -e "${PLYMOUTH_LOGO_FILE}" && inst_simple "${PLYMOUTH_LOGO_FILE}"
 
