--- dracut-017/modules.d/./50plymouth/module-setup.sh~	2012-03-23 00:34:27.000000000 +0200
+++ dracut-017/modules.d/./50plymouth/module-setup.sh	2012-03-23 00:34:29.115606192 +0200
@@ -35,12 +35,12 @@
 }
 
 install() {
-    if grep -q nash /usr/libexec/plymouth/plymouth-populate-initrd \
-        || [ ! -x /usr/libexec/plymouth/plymouth-populate-initrd ]; then
+    if grep -q nash /usr/@lib@/plymouth/plymouth-populate-initrd \
+        || [ ! -x /usr/@lib@/plymouth/plymouth-populate-initrd ]; then
         . "$moddir"/plymouth-populate-initrd.sh
     else
         PLYMOUTH_POPULATE_SOURCE_FUNCTIONS="$dracutfunctions" \
-            /usr/libexec/plymouth/plymouth-populate-initrd -t $initdir
+            /usr/@lib@/plymouth/plymouth-populate-initrd -t $initdir
     fi
 
     inst_hook pre-pivot 90 "$moddir"/plymouth-newroot.sh
