--- dracut-ng-106/modules.d/45plymouth/module-setup.sh.orig	2025-02-20 00:44:56.963325756 +0100
+++ dracut-ng-106/modules.d/45plymouth/module-setup.sh	2025-02-20 00:46:23.059992410 +0100
@@ -1,9 +1,7 @@
 #!/bin/bash
 
 pkglib_dir() {
-    local _dirs="/usr/lib/plymouth /usr/libexec/plymouth/"
-    local _arch=${DRACUT_ARCH:-$(uname -m)}
-    [ -n "$_arch" ] && _dirs+=" /usr/lib/$_arch/plymouth"
+    local _dirs="@libexecdir@/plymouth/"
     for _dir in $_dirs; do
         if [ -x "$dracutsysrootdir""$_dir"/plymouth-populate-initrd ]; then
             echo "$_dir"
