--- dracut-020/modules.d/50plymouth/plymouth-populate-initrd.sh~	2012-08-09 13:49:07.894845601 +0200
+++ dracut-020/modules.d/50plymouth/plymouth-populate-initrd.sh	2012-08-09 13:50:43.211508906 +0200
@@ -6,9 +6,10 @@
 PLYMOUTH_THEME=$(plymouth-set-default-theme)
 
 dracut_install plymouthd plymouth \
-    "${PLYMOUTH_LOGO_FILE}" \
     /etc/os-release
 
+[[ -f "${PLYMOUTH_LOGO_FILE}" ]] && dracut_install "${PLYMOUTH_LOGO_FILE}"
+
 mkdir -m 0755 -p "${initdir}/usr/share/plymouth"
 
 inst_libdir_file "plymouth/text.so" "plymouth/details.so"
