--- dracut-029/dracut.sh~	2013-06-14 15:31:28.000000000 +0200
+++ dracut-029/dracut.sh	2013-06-19 08:29:40.828089414 +0200
@@ -1148,10 +1148,10 @@
 if [[ $UID = 0 ]] && [[ $PRELINK_BIN ]]; then
     if [[ $DRACUT_FIPS_MODE ]]; then
         dinfo "*** Installing prelink files ***"
-        dracut_install -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /etc/prelink.cache
+        dracut_install -l -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /etc/prelink.cache
     else
         dinfo "*** Pre-linking files ***"
-        dracut_install -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf
+        dracut_install -l -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf
         chroot "$initdir" "$PRELINK_BIN" -a
         rm -f "$initdir/$PRELINK_BIN"
         rm -fr "$initdir"/etc/prelink.*
