--- dracut-032/dracut.sh.orig	2013-08-27 15:36:03.692909358 +0200
+++ dracut-032/dracut.sh	2013-08-27 15:39:08.759568281 +0200
@@ -1234,10 +1234,10 @@
 if [[ $UID = 0 ]] && [[ $PRELINK_BIN ]]; then
     if [[ $DRACUT_FIPS_MODE ]]; then
         dinfo "*** Installing prelink files ***"
-        inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /etc/prelink.cache
+        inst_multiple -l -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf /etc/prelink.cache
     else
         dinfo "*** Pre-linking files ***"
-        inst_multiple -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf
+        inst_multiple -l -o prelink /etc/prelink.conf /etc/prelink.conf.d/*.conf
         chroot "$initdir" "$PRELINK_BIN" -a
         rm -f -- "$initdir/$PRELINK_BIN"
         rm -fr -- "$initdir"/etc/prelink.*
