--- dracut-018/modules.d/99shutdown/shutdown.sh~	2012-04-05 13:54:38.000000000 +0200
+++ dracut-018/modules.d/99shutdown/shutdown.sh	2012-04-16 14:04:49.481733793 +0200
@@ -98,7 +98,7 @@
 _check_shutdown final
 
 getarg 'rd.break=shutdown' && emergency_shell -n cmdline "Break before shutdown"
-[ "$1" = "reboot" ] && reboot -f -d -n --no-wall
-[ "$1" = "poweroff" ] && poweroff -f -d -n --no-wall
-[ "$1" = "halt" ] && halt -f -d -n --no-wall
+[ "$1" = "reboot" ] && reboot -f -d -n
+[ "$1" = "poweroff" ] && poweroff -f -d -n
+[ "$1" = "halt" ] && halt -f -d -n
 [ "$1" = "kexec" ] && kexec -e
