--- dracut-047/modules.d/98dracut-systemd/emergency.service.orig	2018-02-19 19:33:03.992393899 +0100
+++ dracut-047/modules.d/98dracut-systemd/emergency.service	2018-02-19 19:34:10.421940966 +0100
@@ -16,7 +16,7 @@
 Environment=NEWROOT=/sysroot
 WorkingDirectory=/
 ExecStart=/bin/dracut-emergency
-ExecStopPost=-/usr/bin/systemctl --fail --no-block default
+ExecStopPost=-/bin/systemctl --fail --no-block default
 Type=idle
 StandardInput=tty-force
 StandardOutput=inherit
